File: 1047.删除字符串中的所有相邻重复项/solution.cpp
class Solution {
public:
    string removeDuplicates(string S) {
        vector<char> ans;
        ans.reserve(S.size());
        for (char i:S) {
            ans.push_back(i);
            int sz=ans.size();
            while (sz>=2 && ans.at(sz-1)==ans.at(sz-2)) {
                ans.pop_back();
                ans.pop_back();
                sz-=2;
            }
        }
        return string(ans.begin(),ans.end());
    }
};

File: 1047.删除字符串中的所有相邻重复项/solution.go
func removeDuplicates(S string) string {
	ans := make([]rune, 0, len(S))
	for _, i := range S {
		ans = append(ans, i)
		sz := len(ans)
		for sz >= 2 && ans[sz-1] == ans[sz-2] {
			ans = ans[:sz-2]
			sz -= 2
		}
	}
	return string(ans)
}
